### DISCUSSION TITLE
Question About "Milvus does not support adding the coordinator components."

### DISCUSSION THREAD

**Discussion by Imccccc** [milvus-io/milvus#24276] [Q&A and General discussion]
Milvus文档中提到"Milvus only supports adding the worker nodes and does not support adding the coordinator components.
"，那么coordinator service中root，query，Data，index4个模块都不能改变

请问coordinate service中几个模块必须单点还是支持多点？
如果支持多点部署
2.1 只能在Milvus部署初期确定，如果集群规模变大出现瓶颈，只能迁移数据到coordinator service更大的集群？
2.2 有官方参考值吗？
如果只支持单点部署
3.1 如果出现性能瓶颈，只能更换更加强的coordinator service?
The Milvus documentation mentions, "Milvus only supports adding the worker nodes and does not support adding the coordinator components." Therefore, the root, query, Data, and index modules in the coordinator service cannot be changed.

I would like to inquire whether the several modules in the coordinator service must be deployed as single nodes or support multiple nodes.
If multiple node deployment is supported:
2.1 Can it only be determined during the initial deployment of Milvus? If the cluster scales up and encounters bottlenecks, is it only possible to migrate the data to a larger coordinator service cluster?
2.2 Are there any official reference values available?
If only single node deployment is supported:
3.1 If there is a performance bottleneck, can the coordinator service only be replaced with a more powerful one?

**Comment by xiaofan-luan** [comment]
> Milvus文档中提到"Milvus only supports adding the worker nodes and does not support adding the coordinator components. "，那么coordinator service中root，query，Data，index4个模块都不能改变
> 
> 请问coordinate service中几个模块必须单点还是支持多点？ 如果支持多点部署 2.1 只能在Milvus部署初期确定，如果集群规模变大出现瓶颈，只能迁移数据到coordinator service更大的集群？ 2.2 有官方参考值吗？ 如果只支持单点部署 3.1 如果出现性能瓶颈，只能更换更加强的coordinator service? The Milvus documentation mentions, "Milvus only supports adding the worker nodes and does not support adding the coordinator components." Therefore, the root, query, Data, and index modules in the coordinator service cannot be changed.
> 
> I would like to inquire whether the several modules in the coordinator service must be deployed as single nodes or support multiple nodes. If multiple node deployment is supported: 2.1 Can it only be determined during the initial deployment of Milvus? If the cluster scales up and encounters bottlenecks, is it only possible to migrate the data to a larger coordinator service cluster? 2.2 Are there any official reference values available? If only single node deployment is supported: 3.1 If there is a performance bottleneck, can the coordinator service only be replaced with a more powerful one?

1.  you are always scale coordinator with more physical resources, for example, change it from 1c4g -> 2c8g.
2. coordinator service can be a single point or primary/backup mode.
3. I would suggest to start from 1c4g, unless your deployment is huge
4. Usually there be no bottleneck on coordinator service . If there is you can scale coordinator up.
